homeassistant/components/switchbot/config_flow.py

```diff
@@ -265,6 +268,91 @@ async def async_step_lock_key(
             },
         )
 
+    async def async_step_choose_method(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the SwitchBot API chose method step."""
+        assert self._discovered_adv is not None
+
+        return self.async_show_menu(
+            step_id="choose_method",
+            menu_options=["auth", "key"],
+            description_placeholders={
+                "name": name_from_discovery(self._discovered_adv),
+            },
+        )
+
+    async def async_step_auth(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the SwitchBot API auth step."""
+        errors = {}
+        assert self._discovered_adv is not None
+        description_placeholders = {}
+        if user_input is not None:
+            try:
+                key_details = await SwitchbotLock.async_retrieve_encryption_key(
```

Comment:
Its a bit unexpected that `SwitchbotLock` is being called here. I think we move around the method in the lib but keep the old one working as well for back-compat.

Reply:
We should also de-duplicate this as it looks like `async_step_lock_auth` is almost the same